packages/passwordless-auth-construct/src/custom-auth/custom_auth_service.ts

```diff
@@ -187,9 +192,38 @@ export class CustomAuthService {
 
     const method = this.validateSignInMethod(signInMethod);
 
-    return this.challengeServiceFactory
+    const verifyResult = this.challengeServiceFactory
       .getService(method)
       .verifyChallenge(event);
+
+    const answerCorrect = (await verifyResult).response.answerCorrect;
+    logger.debug(`Answer is correct: ${answerCorrect ? 'true' : 'false'}`);
+
+    const userCreatedByPasswordlessSignUp =
+      event.request.userAttributes[PASSWORDLESS_SIGNUP_ATTR_NAME] === 'true';
+    const attributeName =
+      event.request.privateChallengeParameters.deliveryMedium === 'SMS'
+        ? 'phone_number_verified'
+        : 'email_verified';
```

Comment:
The value from `privateChallengeParameters` cannot be used. It will not be there in the magic link flow since a new custom auth session is started for link verification.

The delivery medium (or attribute name) should be stored as the value of the custom attribute.